wallet/test/wallet.go

```diff
@@ -89,18 +85,29 @@ func TestAccountWithWalletAndBackend(t *testing.T, s *Setup) { //nolint:revive /
 	buff := new(bytes.Buffer)
 	err = perunio.Encode(buff, sig)
 	require.NoError(t, err, "encode sig")
-	sign2, err := s.Backend.DecodeSig(buff)
+	sign2 := s.Backend.NewSig()
+	err = perunio.Decode(buff, sign2)
 	assert.NoError(t, err, "Decoded signature should work")
 	assert.Equal(t, sig, sign2, "Decoded signature should be equal to the original")
 
 	// Test DecodeSig on short stream
 	err = perunio.Encode(buff, sig)
 	require.NoError(t, err, "encode sig")
 	shortBuff := bytes.NewBuffer(buff.Bytes()[:len(buff.Bytes())-1]) // remove one byte
-	_, err = s.Backend.DecodeSig(shortBuff)
+	sign3 := s.Backend.NewSig()
+	err = perunio.Decode(shortBuff, sign3)
 	assert.Error(t, err, "DecodeSig on short stream should error")
 }
 
+func marshalAndTamper(t *testing.T, sig wallet.Sig, tamperer func(sigBytes *[]byte)) []byte {
+	t.Helper()
+
+	sigBytes, err := sig.MarshalBinary()
+	require.NoError(t, err)
+	tamperer(&sigBytes)
+	return sigBytes
+}
+
 // GenericSignatureSizeTest tests that the size of the signatures produced by
 // Account.Sign(…) does not vary between executions (tested with 2048 samples).
 func GenericSignatureSizeTest(t *testing.T, s *Setup) {
```

Comment:
I think we can throw that test out. Where do we specify that a signature byte representation must always have the same length?